src/screens/WordsScreen/WordsScreen.js

```diff
@@ -14,37 +20,57 @@ class WordsScreen extends Component {
       navigation: { navigate, },
     } = this.props
 
-    navigate('WordDetails', { id, word, })
+    navigate(WORDS_DETAILS_SCREEN, { id, word, })
   }
 
-  toNewWord = () => {
+  toNewWord = (word) => {
     const {
       navigation: { navigate, },
     } = this.props
 
-    navigate('NewWord')
+    if (!word) {
+      navigate(NEW_WORD_SCREEN)
+    } else {
+      navigate(NEW_WORD_SCREEN, { newWord: word, })
+    }
   }
 
-  addNewWord = (word) => {
-    const {
-      navigation: { navigate, },
-    } = this.props
+  setPermissionFunctions = (resolve) => {
+    this.setState({
+      permissionVisible: true,
+      permissionResolve: resolve,
+    })
+  }
 
-    navigate('NewWord', { newWord: word, })
+  refreshPermission = () => {
+    this.setState({
+      permissionVisible: false,
+      permissionResolve: null,
+    })
   }
 
   render() {
+    const { permissionVisible, permissionResolve, } = this.state
     const { container, addButton, } = styles
```

Comment:
Styles destructuring

Reply:
should I  not use the destructuring with styles even they take only one line and they are comfortable for reading?